src/opamParserTypes.mli

```diff
@@ -10,61 +10,124 @@
 (**************************************************************************)
 
 (** Defines the types for the opam format lexer and parser *)
+module Common: sig
 
-(** Relational operators *)
-type relop = [ `Eq  (** [=] *)
-             | `Neq (** [!=] *)
-             | `Geq (** [>=] *)
-             | `Gt  (** [>] *)
-             | `Leq (** [<=] *)
-             | `Lt  (** [<] *)
-             ]
+  (** Relational operators *)
+  type relop = [ `Eq  (** [=] *)
+               | `Neq (** [!=] *)
+               | `Geq (** [>=] *)
+               | `Gt  (** [>] *)
+               | `Leq (** [<=] *)
+               | `Lt  (** [<] *)
+               ]
 
-(** Logical operators *)
-type logop = [ `And (** [&] *) | `Or (** [|] *) ]
+  (** Logical operators *)
+  type logop = [ `And (** [&] *) | `Or (** [|] *) ]
 
-(** Prefix operators *)
-type pfxop = [ `Not (** [!] *) | `Defined (** [?] *) ]
+  (** Prefix operators *)
+  type pfxop = [ `Not (** [!] *) | `Defined (** [?] *) ]
 
-type file_name = string
+  type file_name = string
 
-(** Source file positions: [(filename, line, column)] *)
-type pos = file_name * int * int
+  (** Source file positions: [(filename, line, column)] *)
+  type pos = file_name * int * int
 
-(** Environment variable update operators *)
-type env_update_op = Eq       (** [=] *)
-                   | PlusEq   (** [+=] *)
-                   | EqPlus   (** [=+] *)
-                   | ColonEq  (** [:=] *)
-                   | EqColon  (** [=:] *)
-                   | EqPlusEq (** [=+=] *)
+  (** Environment variable update operators *)
+  type env_update_op = Eq       (** [=] *)
+                     | PlusEq   (** [+=] *)
+                     | EqPlus   (** [=+] *)
+                     | ColonEq  (** [:=] *)
+                     | EqColon  (** [=:] *)
+                     | EqPlusEq (** [=+=] *)
+end
+
+module FullPos : sig
+
+  (** Source file positions *)
+  type file_name = Common.file_name
+  type pos =
+    { filename: file_name;
+      start: int * int; (* line, column *)
+      stop: int * int; (* line, column *)
+    }
+  type 'a with_pos =
+    { pelem : 'a;
+      pos : pos
+    }
+
+  type relop_kind = Common.relop
+  and rel_op = relop_kind with_pos
+
+  type logop_kind = Common.logop
+  and log_op = logop_kind with_pos
+
+  type pfxop_kind = Common.pfxop
+  and pfx_op = pfxop_kind with_pos
+
+  type env_update_op_kind = Common.env_update_op
+  and env_update_op = env_update_op_kind with_pos
```

Comment:
This doesn't work as intended without listing the constructors - I presume it's meant to be that we have `OpamParserTypes.FullPos.Eq`, etc.?). This creates a type alias, not an equivalent type, so you can have:

```ocaml
let x : OpamParserTypes.FullPos.env_update_op_kind = OpamParserTypes.Eq
(* or *)
let x : OpamParserTypes.FullPos.env_update_op_kind = OpamParserTypes.Common.Eq
```
but not
```ocaml
let x : OpamParserTypes.FullPos.env_update_op_kind = OpamParserTypes.FullPos.Eq
```
so:
```suggestion
  type env_update_op_kind = env_update_op =
    Eq | PlusEq | EqPlus | ColonEq | EqColon | EqPlusEq
  type env_update_op = env_update_op_kind with_pos
```